### DISCUSSION TITLE
renovate IgnoreTests with GitLab merges before pipeline

### DISCUSSION THREAD

**Discussion by dkrowl** [renovatebot/renovate#19250] [Request Help] [answered]
I have the following configuration:

```json
"extends": ["group:monorepos","group:recommended","workarounds:all"],
"gitLabAutomerge": true,
  "packageRules": [
    {
      "matchUpdateTypes": ["minor", "patch"],
      "automerge": true
    }
```

The problem is when this configuration is applied to project without tests in Gitlab, automerge won't work.
If I disable the test with: `"ignoreTests": true` then:
* Renovate creted branch is immediately merged to default branch and the source branch is deleted.
* Renovate created dependency branch is started in the pipeline and immediately fails because the source branch has been deleted already.

I'm sure this isn't supposed to work like this. Renovate should wait for the created branches pipeline to succeed and after that gitlabAutomerge should merge the source to target branch. Right?

**Comment by rarkins** [comment, marked as answer]
Do you want Renovate to do the automerging, or GitLab to do it? If GitLab does it, I'm not sure there's a way to do that if you have no tests, because doesn't GitLab wait for at least one test to be present?

**Reply by dkrowl** [reply]
you're right. I totally have mixed two functionalities. But if we forget the gitlab automerge.

I think Renovate should do the merge on the next run if pipeline has succeeded when the `"ignoreTests": true` is applied. And not immediately.

**Reply by rarkins** [reply]
It's functioning as intended today. Renovate's `ignoreTests` means ignore tests completely, so if `automerge` is also enabled then it means automerge the update immediately. `ignoreTests` definitely does not mean "merge on the next run if pipeline has succeeded".
